kitsune/wiki/utils.py

```diff
@@ -182,26 +182,54 @@ def get_visible_document_or_404(
     except Document.DoesNotExist:
         pass
 
-    if (
-        not look_for_translation_via_parent
-        or not (locale := kwargs.get("locale"))
-        or (locale == settings.WIKI_DEFAULT_LANGUAGE)
-    ):
-        # We either don't want to try to find the translation via its parent, or it doesn't
-        # make sense, because we're not making a locale-specific request or the locale we're
-        # requesting is already the default locale.
+    if not (locale := kwargs.get("locale")):
         raise Http404
 
-    # We couldn't find a visible translation in the requested non-default locale, so let's
-    # see if we can find a visible translation via its parent.
+    slug = kwargs.get("slug")
+
+    # For cross-locale lookup
+    if slug and look_for_translation_via_parent:
+        if locale == settings.WIKI_DEFAULT_LANGUAGE:
+            # Find documents with this slug in other locales
+            other_docs_qs = Document.objects.filter(slug=slug).exclude(locale=locale)
```

Comment:
You are repeating this code in the `else` clause too. This can be a bit more DRY by eliminating the second for loop and have everything in this one

Reply:
Adding here a `select_related` for the parent can also help with DB performance

Reply:
Ah - got it. Thanks!